src/Modules/UngradedModules/ModuleGens.jl

```diff
@@ -263,19 +263,19 @@ Convert a Singular vector to a free module element.
 """
 function (F::FreeMod)(s::Singular.svector)
   pos = Int[]
-  values = []
   Rx = base_ring(F)
-  R = base_ring(Rx)
-  for (i, e, c) = s
-    f = Base.findfirst(==(i), pos)
+  R = coefficient_ring(Rx)
+  values = elem_type(Rx)[]
+  for (i, e, c) in s
+    f = Base.findfirst(x->x==i, pos)
     if f === nothing
-      push!(values, MPolyBuildCtx(base_ring(F)))
+      push!(values, zero(Rx))
```

Comment:
Were these deliberately reverted because we do not have an `MPolyBuildCtx` over quotient rings? In that case, I suggest to refactor this part of the code via type dispatch. We talked about it during last week's workshop how to speed this up, eventually. Also, I think @Lax202 's work in progress on non-commutative groebner bases might touch this part again. So we should eventually decide how to split this up here.

Reply:
Follow up: I tried do speed up some of our computations yesterday and made a patch for this part here in #4147 . Would it be desirable to push it directly to this PR? If so, please let me know.

Reply:
Thanks for taking a look and fixing this! Merging this in unison would be good for me.